ykutil/src/addr.rs

```diff
@@ -26,16 +40,18 @@ pub fn code_vaddr_to_off(vaddr: usize) -> Option<(&'static CStr, u64)> {
     if unsafe { dladdr(vaddr as *const c_void, &mut info as *mut Dl_info) } == 0 {
         return None;
     }
-    let containing_obj = unsafe { CStr::from_ptr(info.dli_fname) };
+    let containing_obj = canonicalise_path(unsafe { CStr::from_ptr(info.dli_fname) });
 
     // Find the corresponding byte offset of the virtual address in the object.
     for obj in &objects() {
-        let mut obj_name = obj.name();
-        let obj_name_p = obj_name.as_ptr();
-        if unsafe { *obj_name_p } == 0 {
+        let obj_name = obj.name();
+        let obj_name: PathBuf = if unsafe { *obj_name.as_ptr() } == 0 {
             // On some systems, the empty string indicates the main binary.
-            obj_name = CUR_EXE_C.as_c_str();
-        }
+            // Note that the result of current_exe() is already canonical.
```

Comment:
Instead of this comment how about something like `debug_assert_eq!(env::current_exe(), canonicalise_path(env::curent_exe());`?